modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/MessageExpirationWorker.java

```diff
@@ -62,8 +81,72 @@ public void run() {
+
+                        for (String queueName : queues) {
+                            //get the upper bound messageID for each unassigned slots as a set for the specific queue
+                            TreeSet<Long> messageIDSet = andesContextStore.getMessageIds(queueName);
+
+                            /**
+                             * deletion task run only if there are more messages than the safety slot count upper bound
+                             * otherwise deletion task is not run
+                             * minimum safetySlotCount is 1
+                             */
+                            if (safetySlotCount >= 1 && messageIDSet.size() >= safetySlotCount) {
+
+                                //set the lower bound Id for safety delete region as the safety slot count interval upper bound id + 1
+                                long currentDeletionRangeLowerBoundId = messageIDSet.
+                                        toArray(new Long[messageIDSet.size()])[safetySlotCount - 1] + 1;
+
+                                //get the expired messages for that queue in the range of message ID starting form the lower bound ID
+                                List<AndesMessageMetadata> expiredMessages = MessagingEngine.getInstance()
+                                        .getExpiredMessages(currentDeletionRangeLowerBoundId, queueName);
+
+                                if (null != expiredMessages && !expiredMessages.isEmpty()) {
+
+                                    //in order to stop the slot delivery worker to this queue in the deletion range
+                                    SlotManagerClusterMode.setCurrentDeletionQueue(queueName);
+                                    SlotManagerClusterMode.setCurrentDeletionRangelowerboundID(currentDeletionRangeLowerBoundId);
+                                    SlotManagerStandalone.setCurrentDeletionQueue(queueName);
+                                    SlotManagerStandalone.setCurrentDeletionRangelowerboundID(currentDeletionRangeLowerBoundId);
+
+                                    //delete message metadata, content from the db
+                                    MessagingEngine.getInstance().deleteMessages(expiredMessages);
+
+                                    //since the deletion task is finished that queue no need to hold the slot delivery worker for that queue
+                                    SlotManagerClusterMode.setCurrentDeletionQueue("");
+                                    SlotManagerClusterMode.setCurrentDeletionRangelowerboundID(0L);
+                                    SlotManagerStandalone.setCurrentDeletionQueue("");
+                                    SlotManagerStandalone.setCurrentDeletionRangelowerboundID(0L);
+
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Expired message count for queue : " + queueName + "is" + expiredMessages.size());
```

Comment:
This method is too long it seems. what are the chances of refactoring into smaller methods etc?